include/osquery/registry.h

```diff
@@ -221,7 +229,9 @@ class RegistryInterface : private boost::noncopyable {
   virtual PluginRef plugin(const std::string& plugin_name) const = 0;
 
   /// Construct and return a map of plugin names to their implementation.
-  const std::map<std::string, PluginRef>& plugins() {
+  std::map<std::string, PluginRef> plugins() {
```

Comment:
I think this is OK, and it will definitely protect the critical members. However the lifetime and state (enabled, able to call, disabled, etc) of plugins is managed by the registry, so copying out a `PluginRef` is not ideal in the philosophical sense. Practically this is fine and I say #shipit.

Reply:
I agree with you here but there are two things that made me implement it like this: (1) we still have plugin() which will still allow you to get a reference; (2) the way the consumers of this function are implemented they still need to be passed a reference to the plugin and are free to do whatever they want with it. So while I think we should revisit this, for the time being I don't think this makes it conceptually worse and fixes the crashes, so I would go with it.